packages/request-client.js/src/http-metamask-data-access.ts

```diff
@@ -106,26 +103,26 @@ export default class HttpMetaMaskDataAccess extends HttpDataAccess {
     } = await axios.post('/ipfsAdd', { data: block }, this.axiosConfig);
 
     // get the fee required to submit the hash
-    const fee = await this.submitterContract.getFeesAmount(ipfsSize);
+    const fee = await submitterContract.getFeesAmount(ipfsSize);
 
     // submit the hash to ethereum
-    const tx = await this.submitterContract.submitHash(
+    const tx = await submitterContract.submitHash(
       ipfsHash,
       /* eslint-disable no-magic-numbers */
       ethers.utils.hexZeroPad(ethers.utils.hexlify(ipfsSize), 32),
       { value: fee },
     );
 
-    const ethBlock = await this.provider.getBlock(tx.blockNumber);
+    const ethBlock = await this.provider.getBlock(tx.blockNumber || -1);
```

Comment:
```suggestion
    const ethBlock = await this.provider.getBlock(tx.blockNumber ?? -1);
```
Any reason for not sticking with the nullish coalescing? It has impacts in edge cases like blockNumber = 0. Even if unlikely, I thought we should stick with the good practice.